keps/sig-scheduling/5004-dra-extended-resource/README.md

```diff
@@ -0,0 +1,1132 @@
+
+No. The this KEP proposes extensions to an existing type, but not a new type itself.
+
+###### Will enabling / using this feature result in any new calls to the cloud provider?
+
+No.
+
+###### Will enabling / using this feature result in increasing size or count of the existing API objects?
+
+Yes. With the extensions proposed in this KEP, individual
+`DeviceClass`  and `Pod` have additional fields, thus increasing
+their overall signature. In addition,  there is the special resource claim for
+extended resource by DRA, there is at most one such claim per pod.
+
+###### Will enabling / using this feature result in increasing time taken by any operations covered by existing SLIs/SLOs?
+
+Yes. The time to allocate a device to a pod with extended resource request will be affected.
+
+###### Will enabling / using this feature result in non-negligible increase of resource usage (CPU, RAM, disk, IO, ...) in any components?
+
+No.
+
+###### Can enabling / using this feature result in resource exhaustion of some node resources (PIDs, sockets, inodes, etc.)?
+
+No.
+
+### Troubleshooting
+
+<!--
+This section must be completed when targeting beta to a release.
+
+For GA, this section is required: approvers should be able to confirm the
+previous answers based on experience in the field.
+
+The Troubleshooting section currently serves the `Playbook` role. We may consider
+splitting it into a dedicated `Playbook` document (potentially with some monitoring
+details). For now, we leave it here.
+-->
+
+###### How does this feature react if the API server and/or etcd is unavailable?
+
+Will be considered for beta.
+
+###### What are other known failure modes?
+
+<!--
+For each of them, fill in the following information by copying the below template:
+  - [Failure mode brief description]
+    - Detection: How can it be detected via metrics? Stated another way:
+      how can an operator troubleshoot without logging into a master or worker node?
+    - Mitigations: What can be done to stop the bleeding, especially for already
+      running user workloads?
+    - Diagnostics: What are the useful log messages and their required logging
+      levels that could help debug the issue?
+      Not required until feature graduated to beta.
+    - Testing: Are there any tests for failure mode? If not, describe why.
+-->
+  - [Pod pending due to extended resource backed by DRA requests no less than 128 devices]
+    - Detection: inspect pod status 'Pending'
+    - Mitigations: reduce the number of devices requested in one extended resource backed by DRA requests
+    - Diagnostics: scheduler logs at level 5 show the reason for the scheduling failure.
+    - Testing: Will be considered for beta.
+
+###### What steps should be taken if SLOs are not being met to determine the problem?
+
+Will be considered for beta.
+
+## Implementation History
+
+- Kubernetes 1.34: KEP accepted.
+
+## Drawbacks
+
+It adds complexity to the scheduler.
+
+## Alternatives
```

Comment:
Option 3: Dra driver implement the extended resources too?

Reply:
it looks way simpler of the dra driver implements the extended resources, and announces both via dra and the extended plugin

Reply:
we discussed this option in the early iteration of this KEP, we felt it's better to keep the change in k/k than on the driver, so that it is implemented once in k/k, rather than once per driver, which is more work on the driver developer.

Reply:
If a DRA driver is also an Device Plugin, then to the scheduler the same device is advertised twice and might get allocated for one pod via extended resources and another pod via ResourceClaim -> racy, :boom: